.github/workflows/analyze.yml

```diff
@@ -0,0 +1,33 @@
+name: Analyze
+on:
+  pull_request:
+  push:
+    branches:
+      - confluence
+      - next
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      security-events: write
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          # Needed for commands that depend on git tags
+          fetch-depth: 0
+      - name: Install dependencies
+        run: |
+          yarn
```

Comment:
you can also avoid the installation step on each run of this action by using a cache restoration:

https://github.com/actions/cache/blob/main/examples.md#node---yarn